### PR TITLE
Instrument metadata proxy

### PR DESCRIPTION
**Pull request by ikehz**
Ref kubernetes/kubernetes#8867.

### CODE REVIEW

**Comment by ikehz** [comment]
Xref https://github.com/kubernetes/kubernetes/pull/54653.

File: metrics/metrics.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	RequestCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "request_count",
+			Help: "Counter of metadata proxy requests broken out by filter result of request and HTTP response code.",
```
Comment (crassirostris) [inline metrics/metrics.go]: `Counter` -> `Number`? Usually it's named like this

`...requests broken down by...`?

File: metadata_proxy.go [hunk 2]
```diff
@@ -56,11 +69,30 @@ func (x xForwardedForStripper) RoundTrip(req *http.Request) (*http.Response, err
...
+	metrics.RequestCounter.WithLabelValues(m.filterResult, strconv.Itoa(code)).Inc()
+	m.ResponseWriter.WriteHeader(code)
+}
+
 type metadataHandler struct {
 	proxy *httputil.ReverseProxy
 }
 
-func newMetadataHandler() http.Handler {
+func newMetadataHandler() *metadataHandler {
```
Comment (crassirostris) [inline metadata_proxy.go:95]: Why do you need to specify type here?

**Review by crassirostris — APPROVED** [review]
LGTM, squash commits?

File: metadata_proxy.go (hunk 2 above)
Comment (ikehz) [reply metadata_proxy.go:95]: No immediate reason, but good practice to accept interfaces, return concrete types.  https://www.reddit.com/r/golang/comments/5cos3x/accept_interfaces_return_structs/

File: metrics/metrics.go (hunk 1 above)
Comment (ikehz) [reply metrics/metrics.go]: Done.

**Pull request merged**